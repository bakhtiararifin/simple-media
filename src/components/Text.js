import React from 'react'
import { Text } from 'react-native-elements'

import { colors } from '@theme'

const CustomText = ({ bold = false, secondary = false, style = {}, ...otherProps }) => {
  const defaultStyle = {
    color: colors.textPrimary,
    fontSize: 14,
    paddingVertical: 5,
  }

  if (bold) {
    defaultStyle.fontWeight = 'bold'
  }

  if (secondary) {
    defaultStyle.color = colors.textSecondary
    defaultStyle.fontSize = 12
  }

  return <Text style={{ ...defaultStyle, ...style }} {...otherProps} />
}

export default CustomText
