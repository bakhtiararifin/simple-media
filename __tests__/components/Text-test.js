import React from 'react'
import Text from '@components/Text'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(<Text>text</Text>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('bold text', () => {
  const tree = renderer.create(<Text bold>text</Text>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('secondary text', () => {
  const tree = renderer.create(<Text secondary>text</Text>).toJSON()
  expect(tree).toMatchSnapshot()
})
