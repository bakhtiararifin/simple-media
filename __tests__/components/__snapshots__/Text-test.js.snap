// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bold text 1`] = `
<Text
  style={
    Array [
      Object {},
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      Object {
        "color": "#777",
        "fontSize": 14,
        "fontWeight": "bold",
        "paddingVertical": 5,
      },
    ]
  }
>
  text
</Text>
`;

exports[`renders correctly 1`] = `
<Text
  style={
    Array [
      Object {},
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      Object {
        "color": "#777",
        "fontSize": 14,
        "paddingVertical": 5,
      },
    ]
  }
>
  text
</Text>
`;

exports[`secondary text 1`] = `
<Text
  style={
    Array [
      Object {},
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      Object {
        "color": "#999",
        "fontSize": 12,
        "paddingVertical": 5,
      },
    ]
  }
>
  text
</Text>
`;
